### PR TITLE
Await session deleteAll() in test teardowns - REFAPP-101

### PR DESCRIPTION
**Pull request by robmckinnon**
To fix potential race condition, make session `deleteAll()` function `async` and `await` it to finish in test teardowns.

### CODE REVIEW

File: app/persistence.js [hunk 1]
```diff
@@ -46,7 +48,8 @@ const store = (() => {
     client.keys('*', cbk);
   };
 
-  const deleteAll = () => client.flushall(noop);
+  const deleteAll = async () =>
+    new Promise(resolve => client.flushall(resolve));
```
Comment (ezodude) [inline app/persistence.js]: 👍

File: app/persistence.js [hunk 2]
```diff
@@ -4,16 +4,18 @@ const url = require('url');
...
 
 if (process.env.REDISTOGO_URL) {
   const rtg = url.parse(process.env.REDISTOGO_URL);
   redisPort = rtg.port;
   redisHost = rtg.hostname;
-  client = redis.createClient(redisPort, redisHost);
+}
+
+const client = redis.createClient(redisPort, redisHost);
+
```
Comment (ezodude) [inline app/persistence.js]: Can we combine the conditional below `if (process.env.REDISTOGO_URL)` with the same conditional above?

Comment (robmckinnon) [reply app/persistence.js]: The conditional below requires `client` be created. So we can't put both together, unless we revert back to original code with `let client` declaration and `client =` in two places.

I changed this code, as originally I tried using `const flushall = promisify(client.flushall)`.

Comment (robmckinnon) [reply app/persistence.js]: But `promisify(client.flushall)` didn't work, as `client.flushall()` never rejects, and this breaks `promisify`.

**Pull request merged**